relay/polkadot/src/coretime_migration.rs

```diff
@@ -0,0 +1,318 @@
+				// we already have executed the migration.
+				Some(key) if key.starts_with(&name_hash) => {
+					log::info!("`MigrateToCoretime` already executed!");
+					return true
+				},
+				// Any other key/no key means that we did not yet have migrated.
+				None | Some(_) => return false,
+			}
+		}
+	}
+}
+
+impl<
+		T: Config + runtime_parachains::dmp::Config,
+		SendXcm: xcm::v4::SendXcm,
+		LegacyLease: GetLegacyLease<BlockNumberFor<T>>,
+	> OnRuntimeUpgrade for MigrateToCoretime<T, SendXcm, LegacyLease>
+{
+	fn on_runtime_upgrade() -> Weight {
+		if Self::already_migrated() {
+			return Weight::zero()
+		}
+
+		log::info!("Migrating existing parachains to coretime.");
+		migrate_to_coretime::<T, SendXcm, LegacyLease>()
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::DispatchError> {
+		if Self::already_migrated() {
+			return Ok(Vec::new())
+		}
+
+		let legacy_paras = paras::Parachains::<T>::get();
+		let config = configuration::ActiveConfig::<T>::get();
+		let total_core_count = config.scheduler_params.num_cores + legacy_paras.len() as u32;
+
+		let dmp_queue_size =
+			crate::dmp::Pallet::<T>::dmq_contents(T::BrokerId::get().into()).len() as u32;
+
+		Ok((total_core_count, dmp_queue_size).encode())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(state: Vec<u8>) -> Result<(), sp_runtime::DispatchError> {
+		if state.is_empty() {
+			return Ok(())
+		}
+
+		log::trace!("Running post_upgrade()");
+
+		let (prev_core_count, prev_dmp_queue_size) = <(u32, u32)>::decode(&mut &state[..]).unwrap();
+
+		let dmp_queue_size =
+			crate::dmp::Pallet::<T>::dmq_contents(T::BrokerId::get().into()).len() as u32;
+		let new_core_count = parachains_assigner_coretime::Pallet::<T>::session_core_count();
+		ensure!(new_core_count == prev_core_count, "Total number of cores need to not change.");
+		ensure!(
+			dmp_queue_size > prev_dmp_queue_size,
+			"There should have been enqueued at least one DMP messages."
+		);
+
+		Ok(())
+	}
+}
+
+// Migrate to Coretime.
+//
+// NOTE: Also migrates `num_cores` config value in configuration::ActiveConfig.
+fn migrate_to_coretime<
+	T: Config,
+	SendXcm: xcm::v4::SendXcm,
+	LegacyLease: GetLegacyLease<BlockNumberFor<T>>,
+>() -> Weight {
+	let legacy_paras = LegacyLease::get_all_parachains_with_leases();
+	let legacy_count = legacy_paras.len() as u32;
+	let now = frame_system::Pallet::<T>::block_number();
+	for (core, para_id) in legacy_paras.into_iter().enumerate() {
+		let r = parachains_assigner_coretime::Pallet::<T>::assign_core(
+			CoreIndex(core as u32),
+			now,
+			vec![(CoreAssignment::Task(para_id.into()), PartsOf57600::FULL)],
+			None,
+		);
+		if let Err(err) = r {
+			log::error!(
+				"Creating assignment for existing para failed: {:?}, error: {:?}",
+				para_id,
+				err
+			);
+		}
+	}
+
+	let config = configuration::ActiveConfig::<T>::get();
+	// num_cores was on_demand_cores until now:
+	for on_demand in 0..config.scheduler_params.num_cores {
```

Comment:
On Polkadot this is not the path of this config entry, checking the chain state it's at `config.coretime_cores`. Is there a migration ahead of this one to change the configuration in storage?

Reply:
`parachains_configuration::migration::v12::MigrateToV12<Runtime>` does this change:
https://github.com/paritytech/polkadot-sdk/blob/ebf4f8d2d590f41817d5d38b2d9b5812a46f2342/polkadot/runtime/parachains/src/configuration/migration/v12.rs#L146-L158

Reply:
I've removed the comment.